### DISCUSSION TITLE
How to test external extension against Theia with local modifications

### DISCUSSION THREAD

**Discussion by simark** [eclipse-theia/theia#9289] [Q&A]
I am trying to work on the package `@theia/cpp-debug`, in [theia-cpp-extension](https://github.com/eclipse/theia-cpp-extension/), which depends on `@theia/debug` in the main Theia repo.  I need to make local modifications to the `@theia/debug` package and test the external extension against that.  How would you do this?

**Comment by simark** [comment]
I tried changing the dependency of `cpp-debug` on `debug` to point to my local build:

```
 23   "dependencies": {
 24     "@theia/core": "next",
 25     "@theia/debug": "link:/home/emaisin/src/theia/packages/debug",
 26     "cdt-gdb-adapter": "link:/home/emaisin/src/cdt-gdb-adapter"
 27   },
```

However, this leads to multiple versions of `@theia/core` to be pulled, so it doesn't work.

**Comment by marcdumais-work** [comment]
I think we could work-around the issue by setting your local development `@theia/debug` to the specific `next` version that the other `@theia` packages will resolve-to. ATM this is : `0.4.0-next.74b3528e`, but it changes every time something merged to master and successfully published.

latest info:
``` npm info @theia/debug```
